torch/csrc/distributed/c10d/init.cpp

```diff
@@ -411,6 +411,38 @@ Example::
     >>> store.set("first_key", "first_value")
     >>> # Should return "first_value"
     >>> store.get("first_key")
+)")
+          .def(
+              "compare_set",
+              [](::c10d::Store& store,
+                 const std::string& key,
+                 const std::string& new_value,
+                 const std::string& old_value) -> py::bytes {
+                std::vector<uint8_t> newValue_(new_value.begin(), new_value.end());
+                std::vector<uint8_t> oldValue_(old_value.begin(), old_value.end());
+                auto value = store.compareSet(key, newValue_, oldValue_);
+                return py::bytes(
```

Comment:
I find it inconsistent that we accept the value as a `std::string` in both `set()` and `compare_set()`, but return a `py::bytes` instance in `get()` and `compare_set()`. I would prefer if we had used `py:bytes` as the parameter type of value as well instead of `std::string`. I am not expecting you to address this issue right now, but this is something we might take a look in the future.

Reply:
Yep, I'm not exactly sure the reason why we convert from string to vector<uint8_t> internally. Perhaps it is to simplify the arguments for the users? I can definitely look into this, thanks!